modules/apps/portal-search-solr8/portal-search-solr8-impl/src/main/java/com/liferay/portal/search/solr8/internal/http/BasePoolingHttpClientFactory.java

```diff
@@ -155,19 +145,14 @@ protected void applyProperties(
 			createPoolingHttpClientConnectionManager()
 		throws Exception;
 
-	protected void removeHttpRequestInterceptor(
-		HttpRequestInterceptor httpRequestInterceptor) {
-
-		_httpRequestInterceptors.remove(httpRequestInterceptor);
-	}
-
 	private static final Log _log = LogFactoryUtil.getLog(
 		BasePoolingHttpClientFactory.class);
 
 	private Integer _defaultMaxConnectionsPerRoute;
-	private final List<HttpRequestInterceptor> _httpRequestInterceptors =
-		new ArrayList<>();
 	private Integer _maxTotalConnections;
 	private PoolingHttpClientConnectionManager _poolingClientConnectionManager;
 
+	@Reference
+	private PreemptiveAuthInterceptor _preemptiveAuthInterceptor;
```

Comment:
@W2jx This can not be right.  You should use protected, so that its sub classes will extend the dependecy.